client/asset/doge/doge.go

```diff
@@ -160,22 +180,36 @@ func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network)
 		SingularWallet:           true,
 		UnlockSpends:             true,
 		ConstantDustLimit:        dustLimit,
-		FeeEstimator: func(cl btc.RawRequester, _ uint64) (uint64, error) {
+		FeeEstimator: func(ctx context.Context, cl btc.RawRequester, _ uint64, allowExternal bool, net dex.Network) (uint64, error) {
 			confArg, err := json.Marshal(feeConfs)
 			if err != nil {
 				return 0, err
 			}
 			resp, err := cl.RawRequest("estimatefee", []json.RawMessage{confArg})
 			if err != nil {
-				return 0, err
+				if !allowExternal {
+					return 0, err
+				}
+				feeRate, err := externalFeeEstimator(ctx, net)
+				if err != nil {
+					return 0, err
+				}
+				return feeRate, nil
 			}
 			var feeRate float64
 			err = json.Unmarshal(resp, &feeRate)
 			if err != nil {
 				return 0, err
 			}
 			if feeRate <= 0 {
-				return 0, nil
+				if !allowExternal {
+					return 0, nil
+				}
+				feeRate, err := externalFeeEstimator(ctx, net)
+				if err != nil {
+					return 0, err
+				}
+				return feeRate, nil
```

Comment:
We can do `return externalFeeEstimator(ctx, net)` for these two places in this closure.  We define the function below, so we know it's going to return 0 for a non-nil error.